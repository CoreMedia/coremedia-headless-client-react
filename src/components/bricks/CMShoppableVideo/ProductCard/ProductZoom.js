// @flow
// import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import theme from 'styled-theming';

import ProductBox from './ProductBox';
import { media } from '../../../../styles/themes/utils';

const hoverStyles = {
  tablet: theme('name', {
    default: css`
      left: -2.5%;
      top: -11.5%;
      width: 105%;
      height: 123%;
    `,
    hkm: css`
      left: -2.5%;
      top: -10%;
      width: 105%;
      height: 120%;
    `,
  }),
  desktop: theme('name', {
    default: css`
      left: -2.5%;
      top: -9.5%;
      width: 105%;
      height: 119%;
    `,
    hkm: css`
      left: -2.5%;
      top: -9%;
      width: 105%;
      height: 118%;
    `,
  }),
  large: theme('name', {
    default: css`
      left: -2.5%;
      top: -9%;
      width: 105%;
      height: 118%;
    `,
    hkm: css`
      left: -2.5%;
      top: -7.5%;
      width: 105%;
      height: 115%;
    `,
  }),
};

const ProductZoom = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: #fff;
  overflow: hidden;
  ${props => media(props.theme.breakpoints.tablet)`
    transition: top 0.1s ease-out, left 0.1s ease-out, width 0.1s ease-out, height 0.1s ease-out;
    ${ProductBox}:hover & {
      z-index: 10;
      box-shadow: 0px 0px 7px 0px rgba(0, 0, 0, 0.32);
    }
  `};
  ${props =>
    ['tablet', 'desktop', 'large'].reduce(
      (acc, breakpoint) =>
        acc.concat(
          media(props.theme.breakpoints[breakpoint])`
            ${ProductBox}:hover & {
              ${hoverStyles[breakpoint]}
            }
          `
        ),
      []
    )};
`;

ProductZoom.displayName = 'ProductZoom';

/*ProductZoom.propTypes = {
  theme: PropTypes.shape({
    name: PropTypes.string.isRequired,
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
  }),
};*/

ProductZoom.defaultProps = {
  theme: {
    name: 'default',
    breakpoints: {
      tablet: 768,
      desktop: 992,
      large: 1280,
    },
  },
};

export default ProductZoom;
