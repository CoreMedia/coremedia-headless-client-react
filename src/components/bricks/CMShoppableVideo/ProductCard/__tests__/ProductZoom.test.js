// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ProductZoom from '../ProductZoom';

describe('ProductZoom Component', () => {
  describe('with default theme', () => {
    it('should render correctly', () => {
      const tree = renderer.create(<ProductZoom>Content</ProductZoom>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('with hkm theme', () => {
    it('should render correctly', () => {
      const tree = renderer
        .create(
          <ProductZoom
            theme={{
              name: 'hkm',
              breakpoints: {
                tablet: 768,
                desktop: 992,
                large: 1280,
              },
            }}
          >
            Content
          </ProductZoom>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
