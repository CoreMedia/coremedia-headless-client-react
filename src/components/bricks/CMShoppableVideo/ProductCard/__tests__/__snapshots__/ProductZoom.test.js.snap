// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductZoom Component with default theme should render correctly 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: #fff;
  overflow: hidden;
}

@media (min-width:48em) {
  .c0 {
    -webkit-transition: top 0.1s ease-out,left 0.1s ease-out,width 0.1s ease-out,height 0.1s ease-out;
    transition: top 0.1s ease-out,left 0.1s ease-out,width 0.1s ease-out,height 0.1s ease-out;
  }

  .sc-bdVaJa:hover .c0 {
    z-index: 10;
    box-shadow: 0px 0px 7px 0px rgba(0,0,0,0.32);
  }
}

@media (min-width:48em) {
  .sc-bdVaJa:hover .c0 {
    left: -2.5%;
    top: -11.5%;
    width: 105%;
    height: 123%;
  }
}

@media (min-width:62em) {
  .sc-bdVaJa:hover .c0 {
    left: -2.5%;
    top: -9.5%;
    width: 105%;
    height: 119%;
  }
}

@media (min-width:80em) {
  .sc-bdVaJa:hover .c0 {
    left: -2.5%;
    top: -9%;
    width: 105%;
    height: 118%;
  }
}

<div
  className="c0"
>
  Content
</div>
`;

exports[`ProductZoom Component with hkm theme should render correctly 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: #fff;
  overflow: hidden;
}

@media (min-width:48em) {
  .c0 {
    -webkit-transition: top 0.1s ease-out,left 0.1s ease-out,width 0.1s ease-out,height 0.1s ease-out;
    transition: top 0.1s ease-out,left 0.1s ease-out,width 0.1s ease-out,height 0.1s ease-out;
  }

  .sc-bdVaJa:hover .c0 {
    z-index: 10;
    box-shadow: 0px 0px 7px 0px rgba(0,0,0,0.32);
  }
}

@media (min-width:48em) {
  .sc-bdVaJa:hover .c0 {
    left: -2.5%;
    top: -10%;
    width: 105%;
    height: 120%;
  }
}

@media (min-width:62em) {
  .sc-bdVaJa:hover .c0 {
    left: -2.5%;
    top: -9%;
    width: 105%;
    height: 118%;
  }
}

@media (min-width:80em) {
  .sc-bdVaJa:hover .c0 {
    left: -2.5%;
    top: -7.5%;
    width: 105%;
    height: 115%;
  }
}

<div
  className="c0"
>
  Content
</div>
`;
